rust/protocol/src/error.rs

```diff
@@ -11,82 +15,208 @@ use uuid::Uuid;
 
 use std::panic::UnwindSafe;
 
+#[cfg(doc)]
+pub use crate::{
+    address::ProtocolAddress,
+    curve::{PrivateKey, PublicKey},
+    group_cipher::{group_decrypt, group_encrypt},
+    protocol::{SenderKeyMessage, SignalMessage, CIPHERTEXT_MESSAGE_CURRENT_VERSION},
+    sealed_sender::{sealed_sender_decrypt, sealed_sender_multi_recipient_encrypt},
+    sender_keys::SenderKeyRecord,
+    session_cipher::{message_decrypt, message_encrypt},
+    state::{PreKeyId, PreKeyRecord, SignedPreKeyId, SignedPreKeyRecord},
+    storage::{IdentityKeyStore, PreKeyStore, SenderKeyStore, SessionStore, SignedPreKeyStore},
+};
+
+/// Return type for all fallible operations in this crate.
 pub type Result<T> = std::result::Result<T, SignalProtocolError>;
 
+/// Error states recognized by the Signal Protocol.
 #[derive(Debug, Display, Error)]
+#[ignore_extra_doc_attributes]
 pub enum SignalProtocolError {
```

Comment:
Heh, this sounds better designed than what's actually going on: the "SignalProtocol" here is really just about the name of the crate, libsignal-protocol. Arguably it could just be Error or something like that; I'd probably document it as "Possible errors produced by the operations in this crate" to match what you did for the Result alias.